registration/api/v2/serializers.py

```diff
@@ -0,0 +1,176 @@
+            user.set_password(user_data.pop('password'))
+            user_updated = True
+        if 'email' in user_data:
+            user.email = user_data.pop('email')
+            user_updated = True
+        if user_updated:
+            user.save()
+
+    def _update_user_profile(self, user_profile: UserProfile, profile_data: Dict):
+        """
+        Update user profile from validated data.
+        """
+        profile_updated = False
+        for key, new_val in profile_data.items():
+            current_val = getattr(user_profile, key)
+            if new_val != current_val:
+                setattr(user_profile, key, new_val)
+                profile_updated = True
+        if profile_updated:
+            user_profile.save()
+
+    def update(self, instance, validated_data):
+        """
+        Update existing user profile model and user.
+        """
+        self._update_user_account(instance.user, validated_data.pop('user', {}))
+        self._update_user_profile(instance, validated_data)
+        return instance
+
+    def validate_accepted_privacy_policy(self, value):
+        """
+        Ensure that no account is created without a policy acceptance date.
+        """
+        if value is None:
+            raise ValidationError("You must accept the privacy policy to register.")
+        if (
+                self.instance and
+                self.instance.accepted_privacy_policy and
+                value < self.instance.accepted_privacy_policy
+        ):
+            raise ValidationError("New policy acceptance date cannot be earlier than previous acceptance date.")
+        if value >= timezone.now() + timezone.timedelta(hours=1):
+            raise ValidationError("Cannot accept policy for a future date.")
+        return value
+
+    def validate_accept_paid_support(self, value):
+        """
+        Ensure that no account exists that hasn't accepted the support terms.
+        """
+        if not value:
+            raise ValidationError("You must accept these terms to register.")
+        return value
+
+    class Meta:
+        model = UserProfile
+        fields = (
+            'full_name', 'username', 'password', 'email', 'accepted_privacy_policy',
+            'accept_paid_support', 'subscribe_to_updates',
+        )
+        extra_kwargs = {
+            'accepted_privacy_policy': {'required': True},
+            'accept_paid_support': {'required': True},
+        }
+
+
+class OpenEdXInstanceConfigSerializer(serializers.ModelSerializer):
+    """
+    Serializer with configuration details about the user's Open edX instance.
+    """
+    user = serializers.HiddenField(default=serializers.CurrentUserDefault())
+
+    def validate_user(self, value):
+        """
+        Prevent user from creating more than the allowed number of instances.
+
+        Currently each user is allowed one Open edX instance.
+        """
+        if BetaTestApplication.objects.filter(user=value).exists():
```

Comment:
We could change `exists` to `count` with and overriddable value for forward compatibility.
We may also just leave it as it is for now and change it in the future, if needed.

Reply:
My understanding is that `exists` is much better for performance if applicable.

Reply:
Yep, it is. Let's use `exists` then, as we currently don't care about multiple instances.